samples/wifi/radio_test/src/nrf_wifi_radio_test_shell.c

```diff
@@ -1189,6 +1199,45 @@ static int nrf_wifi_radio_test_set_tx(const struct shell *shell,
 				      "Invalid channel settings\n");
 			return -ENOEXEC;
 		}
+
+		/*Max TX power values differ based on the test being performed.
+		 *For TX EVM Vs Power, Max TX power required is 24dB
+		 *(96 in 0.25dB resolution) whereas for testing the Max TX power
+		 *for which both EVM and spectrum mask are passing for specific
+		 *band and MCS/rate, TX power values will be read from
+		 *RF params string
+		 */
+		status = wifi_nrf_fmac_rf_params_get(ctx->rpu_ctx,
+					  ctx->conf_params.rf_params);
+
+		if (status != WIFI_NRF_STATUS_SUCCESS) {
+			goto out;
+		}
+
+		if (ctx->conf_params.tx_power == MAX_TX_POWER) {
+			/*Do not change default RF params*/
+		} else {
+			unsigned char max_tx_power = 96;
+
+			ctx->conf_params.rf_params[NRF_WIFI_RF_PARAMS_OFF_CALIB_PWR2G]
+			= max_tx_power;
+			ctx->conf_params.rf_params[NRF_WIFI_RF_PARAMS_OFF_CALIB_PWR2GM0M7]
+			= max_tx_power;
+			ctx->conf_params.rf_params[NRF_WIFI_RF_PARAMS_OFF_CALIB_PWR2GM0M7+1]
+			= max_tx_power;
+			ctx->conf_params.rf_params[NRF_WIFI_RF_PARAMS_OFF_CALIB_PWR5GM7]
+			= max_tx_power;
+			ctx->conf_params.rf_params[NRF_WIFI_RF_PARAMS_OFF_CALIB_PWR5GM7+1]
+			= max_tx_power;
+			ctx->conf_params.rf_params[NRF_WIFI_RF_PARAMS_OFF_CALIB_PWR5GM7+2]
+			= max_tx_power;
+			ctx->conf_params.rf_params[NRF_WIFI_RF_PARAMS_OFF_CALIB_PWR5GM0]
+			= max_tx_power;
+			ctx->conf_params.rf_params[NRF_WIFI_RF_PARAMS_OFF_CALIB_PWR5GM0+1]
+			= max_tx_power;
+			ctx->conf_params.rf_params[NRF_WIFI_RF_PARAMS_OFF_CALIB_PWR5GM0+2]
+			= max_tx_power;
+		}
```

Comment:
Convert this to a helper function which will update RF params based on TX power value.